### PR TITLE
Support SPM

### PR DESCRIPTION
**Pull request by aomathwift**
Supported Swift Package Manager as a means of package distribution.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -5,10 +5,16 @@ jobs:
     runs-on: macos
     steps:
     - uses: actions/checkout@v2
-    - name: Run Tests
+    - name: Run Tests with Xcode
       run: |
         DESTINATION="platform=iOS Simulator,name=iPhone 8" SCHEME="Puree"
         xcodebuild test -project Puree.xcodeproj -scheme "${SCHEME}" -destination "${DESTINATION}"
+  spm:
```
Comment (giginet) [inline .github/workflows/ci.yml]: Prefer to use SwiftPM
(SPM seems to be un-official abbreviation...)

File: Package.swift [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// swift-tools-version:5.0
+// The swift-tools-version declares the minimum version of Swift required to build this package.
+
+import PackageDescription
+
+let package = Package(
+    name: "Puree-Swift",
```
Comment (giginet) [inline Package.swift]: Puree-Swift is repository name.
Package name should be just `Puree`.

File: Package.swift [hunk 3]
```diff
@@ -0,0 +1,34 @@
+// swift-tools-version:5.0
+// The swift-tools-version declares the minimum version of Swift required to build this package.
+
+import PackageDescription
+
+let package = Package(
+    name: "Puree",
+    platforms: [
+        .iOS(.v10),
+        .macOS(.v10_12),
```
Comment (giginet) [inline Package.swift]: We don't support macOS. So you can get rid of this.

File: Package.swift [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        // Dependencies declare other packages that this package depends on.
+        // .package(url: /* package url */, from: "1.0.0"),
+    ],
+    targets: [
+        // Targets are the basic building blocks of a package. A target can define a module or a test suite.
+        // Targets can depend on other targets in this package, and on products in packages which this package depends on.
+        .target(
+            name: "Puree",
+            dependencies: [],
+            path: "Sources/Puree"),
```
Comment (giginet) [inline Package.swift]: It's a default path. So you should not specify `path:`.

File: Package.swift [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        .testTarget(
+            name: "PureeTests",
+            dependencies: ["Puree"],
+            path: "Tests/PureeTests"),
```
Comment (giginet) [inline Package.swift]: same above

File: Package.swift [hunk 6]
```diff
@@ -0,0 +1,34 @@
+// swift-tools-version:5.0
+// The swift-tools-version declares the minimum version of Swift required to build this package.
+
+import PackageDescription
+
+let package = Package(
+    name: "Puree",
+    platforms: [
+        .iOS(.v10),
```
Comment (giginet) [inline Package.swift]: We needn't support such an ancient version 🦖 
Our support policy is not fixed. but it's OK only above .v12.

File: README.md [hunk 7]
```diff
@@ -40,6 +40,16 @@ use_frameworks!
 pod 'Puree', '~> 5.0'
 ```
 
+### Swift PM
+The [Swift Package Manager](https://swift.org/package-manager/) is a tool for automating the distribution of Swift code and is integrated into the `swift` compiler. It is in early development, but Alamofire does support its use on supported platforms.
```
Comment (giginet) [inline README.md]: Alamofire 👀

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -5,10 +5,16 @@ jobs:
...
+    - name: Run Tests with Xcode
       run: |
         DESTINATION="platform=iOS Simulator,name=iPhone 8" SCHEME="Puree"
         xcodebuild test -project Puree.xcodeproj -scheme "${SCHEME}" -destination "${DESTINATION}"
+  SwiftPM:
+    runs-on: macos
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run Tests with Swift Package Manager
+      run: swift test -Xswiftc "-target" -Xswiftc "x86_64-apple-macosx10.12"
```
Comment (aomathwift) [inline .github/workflows/ci.yml:17]: The package manager currently hardcodes the deployment target to macOS 10.10, So need to overwrite the target to pass the test. (I don't know why it doesn't work even though we've specified iOS as a supported platform ...)

Comment (giginet) [reply .github/workflows/ci.yml:17]: According to the documentation, we must specify macOS platform version in current limitation...

 https://github.com/apple/swift-package-manager/blob/03c10bed07709f6b1b2787433ed15001f7ce433c/Sources/PackageDescription/SupportedPlatforms.swift#L52-L66

File: README.md [hunk 9]
```diff
@@ -40,6 +40,16 @@ use_frameworks!
 pod 'Puree', '~> 5.0'
 ```
 
+### Swift PM
+The [Swift Package Manager](https://swift.org/package-manager/) is a tool for automating the distribution of Swift code and is integrated into the `swift` compiler. It is in early development, but Puree-Swift does support its use on supported platforms.
+
+Once you have your Swift package set up, adding Puree-Swift as a dependency is as easy as adding it to the `dependencies` value of your `Package.swift`.
+```swift
+dependencies: [
+    .package(url: "https://github.com/cookpad/Puree-Swift.git", from: "5.1.0")
```
Comment (giginet) [inline README.md]: Prefer to use `upToNextMinor(from:)`

```suggestion
    .package(url: "https://github.com/cookpad/Puree-Swift.git", .upToNextMinor(from: "5.1.0"))
```

**Review by giginet — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 10]
```diff
@@ -1,14 +1,20 @@
 name: Puree
-on: [push]
+on: [pull_request]
```
Comment (giginet) [inline .github/workflows/ci.yml]: `push` should be still supported.

```suggestion
on: [push, pull_request]
```

**Review by giginet — APPROVED** [review]

**Comment by aomathwift** [comment]
Thank you for your review!

**Pull request merged**